test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -2129,6 +2132,30 @@ protected NumShards getNumShards(String index) {
         return nodes;
     }
 
+    /**
+     * Asserts that all shards with the same shardId should have document Ids.
+     */
+    public void assertSameDocIdsOnShards() throws Exception {
```

Comment:
do we need to wrap this in an assertBusy?

Reply:
Done